tardis/plasma/properties/nlte_rate_equation_solver.py

```diff
@@ -848,3 +848,48 @@ def prepare_r_uls_r_lus(
             r_lu_matrix,
         )
         # TODO: beta sobolev needs to be recalculated for each iteration, because it depends on number density
+
+    @staticmethod
+    def create_coll_exc_deexc_matrix(
+        coll_exc_coefficient, coll_deexc_coefficient
+    ):
+        """Generates a coefficient matrix from collisional excitation/deexcitation coefficients.
+
+        Needs to be multiplied by electron density when added to the overall rate_matrix.
+        Parameters
+        ----------
+        coll_exc_coefficient : pandas.DataFrame
+            DataFrame of collisional excitation coefficients for current (atomic number, ion_number)
+        coll_deexc_coefficient : pandas.DataFrame
+            DataFrame of collisional deexcitation coefficients for (atomic number, ion_number)
+
+        Returns
+        -------
+        coeff_matrix : np.array (number of levels, number of levels)
+            Square matrix constructed by collisional exc./deexc. coefficients.
+        """
+        size = (
+            coll_exc_coefficient.index.get_level_values("level_number_lower")
+            .unique()
+            .size
+        )
```

Comment:
Potentially, there might be situations where a level is missing from coll_exc_coefficient because we don't have collisional data for it. It might be safer to just pass the number of levels to the method as an argument. Alternatively, we can make sure that every possible transition is included in coll_exc_coefficient by replacing missing values with zero rates.